From e414d30af53c457f7d6d7e7f230a5e91982302f7 Mon Sep 17 00:00:00 2001
From: zglstudylinux <zglstudylinux@users.noreply.github.com>
Date: Mon, 19 Oct 2026 13:47:23 +0800
Subject: [PATCH] t113: zgl_board enable BT H5 and uart1 four-wire for
 RTL8723DS

- config-5.4: CONFIG_XR_BT_LPM -> CONFIG_BT_HCIUART_RTL3WIRE (H5 protocol;
  XR_BT_LPM is for XR modules)
- board.dts: &uart1 status disabled -> okay (PG6-9 4-wire, BT HCI serial)
---
 configs/zgl_board/linux-5.4/board.dts  | 2 +-
 configs/zgl_board/linux-5.4/config-5.4 | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/configs/zgl_board/linux-5.4/board.dts b/configs/zgl_board/linux-5.4/board.dts
index 7fa7d7a..1e008fd 100644
--- a/configs/zgl_board/linux-5.4/board.dts
+++ b/configs/zgl_board/linux-5.4/board.dts
@@ -811,7 +811,7 @@
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&uart1_pins_a>;
 	pinctrl-1 = <&uart1_pins_b>;
-	status = "disabled";       /* phase-5 (BT) will enable */
+	status = "okay";           /* RTL8723DS BT UART (4-wire, 115200) */
 };
 
 &uart2 {
diff --git a/configs/zgl_board/linux-5.4/config-5.4 b/configs/zgl_board/linux-5.4/config-5.4
index 3cf375d..2b72414 100644
--- a/configs/zgl_board/linux-5.4/config-5.4
+++ b/configs/zgl_board/linux-5.4/config-5.4
@@ -75,7 +75,8 @@ CONFIG_BT_RFCOMM=y
 CONFIG_BT_RFCOMM_TTY=y
 CONFIG_BT_HCIUART=y
 CONFIG_BT_HCIUART_H4=y
-CONFIG_XR_BT_LPM=y
+CONFIG_BT_HCIUART_RTL3WIRE=y
+# CONFIG_XR_BT_LPM is not set
 CONFIG_CFG80211=y
 CONFIG_RFKILL=y
 CONFIG_RFKILL_GPIO=y
-- 
2.43.0

